### DISCUSSION TITLE
Ctrlr versions

### DISCUSSION THREAD

**Discussion by bijlevel** [RomanKubiak/ctrlr#306] [General]
I noticed in the various discussions some confusion about version numbers. Also I think it is a good idea to annotate the various versions. This is my contribution.

For Mac I  still use version **5.3.122**, because with later versions for Mac I did have problems with resources, but for new panels I did not encounter a lot of problems with version **5.3.198**, so I guess this one is OK.
For Windows, I think most people use version **5.3.201**. 

All these versions can be found on the old forum www.ctrlr.org

Roman Kubiak has released various versions 5.5.x, AFAIK only as source and not as installer and they don't get much attention (?).

Recently (April 2021) he released version **5.6.x.** also only as source. This version is now the subject of the various compilings by me. When compiled is says version **5.6.23**. So until a new release this v5.6.23 can be considered as the latest version, but is still being tested. So we can't call it stable yet, I think.

Please correct me if something is wrong or missing, or add your own experiences with a version, so at the end of this thread we have a good oversight of all the different versions ;-).

**Comment by Godlike-Productions** [comment]
6.0.26 was released for windows some time back and seems to have some quirks when rendering VST's under windows. I'm still working out what works and what doesn't.

**Comment by Godlike-Productions** [comment]
I can't get the 5.6.23.exe to run on my computer.  I've tried placing it in it's own directory, the 5.4.29 directory, and the 6.0.26 directory and it fails to run.  No errors, just not running.  I installed the redistributable.

Trying to run the VST3 from VSTHost - it loads fine, but as soon as I try to open the K2x00R Panel I get the following errors

```
Unhandled exception 0xC0000005 at 00007FFAA503FD26 reading from 00000000098E7F44

CS:RIP=33:00007FFAA503FD26 SS:RSP=2B:000000000014EE68 RBP=0000000000000000
Flags=0000000000010203 MXCSR=0000000000001FA5
DS =000000000000002B ES =000000000000002B FS =0000000000000053 GS =000000000000002B
RAX=000000000B204C30 RBX=000000000B192780 RCX=00000000096E0080 RDX=00000000098DD710
RSI=00000000096E0100 RDI=0000000000000000
R8 =0000000000002A0D R9 =000000000E06131E R10=000000000E060000 R11=000000000014EFB8
R12=00000000FFFFFFFF R13=000000002ADA5FE8 R14=0000000000000001 R15=0000000000002A0D
Stack Trace:
00007FFAA503FD26 (Ctrlr-x64): (file?): (function?)
00007FFAA55B4518 (Ctrlr-x64): InitDll + 0x18bfb8
00007FFAA55B9F49 (Ctrlr-x64): InitDll + 0x1919e9
00007FFAA55BDD3D (Ctrlr-x64): InitDll + 0x1957dd
00007FFAA55BEE01 (Ctrlr-x64): InitDll + 0x1968a1
00007FFAA55CDD0E (Ctrlr-x64): InitDll + 0x1a57ae
00007FFAA55CE89A (Ctrlr-x64): InitDll + 0x1a633a
00007FFAA55CC107 (Ctrlr-x64): InitDll + 0x1a3ba7
00007FFAA5668C76 (Ctrlr-x64): InitDll + 0x240716
00007FFAA550EF8C (Ctrlr-x64): InitDll + 0xe6a2c
00007FFAA5093747 (Ctrlr-x64): (file?): (function?)
00007FFAA50933C7 (Ctrlr-x64): (file?): (function?)
00007FFB2C9EE858 (USER32): CallWindowProcW + 0x3f8
00007FFB2CA2F944 (USER32): LoadMenuA + 0x124
00007FFB2CA2FF8B (USER32): CallWindowProcA + 0x1b
00000001401CA8D5 (vsthost): _AfxActivationWndProc + 0x16d
00007FFB2C9EE858 (USER32): CallWindowProcW + 0x3f8
00007FFB2C9EE299 (USER32): DispatchMessageW + 0x259
00000001401DBDEE (vsthost): AfxInternalPumpMessage + 0x52
00000001401DB8AB (vsthost): CWinThread::Run + 0x6f
000000014026F087 (vsthost): AfxWinMain + 0xab
000000014023B411 (vsthost): set_abort_behavior + 0x21d
00007FFB2D717034 (KERNEL32): BaseThreadInitThunk + 0x14
00007FFB2D8A2651 (ntdll): RtlUserThreadStart + 0x21
```

**Reply by Godlike-Productions** [reply]
Tried again and it opened OK this time in the VST3

**Reply by Godlike-Productions** [reply]
I've been messing around a bit with the VST3 you exported.  It seems to work about the same as V6.0.26, but, and it's a big but, the menu bar will not show up, so you can't assign MIDI devices the normal way.

Also the VST3 has a fixed size bounds, so if you are above Zoom 1 it won't fit in the window (well, you can, as long as you don't have an image in the background, as it will stretch to the new bounds, without stretching the layers.

Anyway, have managed to get a working VST3 out for the K2x00 panel.
